python/cudf/cudf/tests/test_dtypes.py

```diff
@@ -257,3 +257,44 @@ def test_lists_of_structs_dtype(data):
 
     assert_column_array_dtype_equal(got._column, expected)
     assert expected.equals(got._column.to_arrow())
+
+
+@pytest.mark.parametrize(
+    "in_dtype,expect",
```

Comment:
We probably want to test more inputs that don't translate to numpy dtypes, specifically more cudf- and pandas-specific extension types.

Reply:
The cuDF specific types (i.e., instances of `cudf._BaseDtype`) are less interesting since we just return those as-is. But I did add a few more tests.

Reply:
Maybe also worth testing pandas interval/datetime/timedelta dtypes.

Reply:
Added interval cases, but as far as I know, Pandas uses numpy datetime/timedelta types as their dtype for `DatetimeIndex`/`TimedeltaIndex`.